Fixes #22907 - expose Pulp DB configuration params

mbacovsky opened the pull request:
This patch exposes most of the Pulp DB configuration and visible from installer.
The params are listed as advanced and not pollute installer --help.
I was able to install Katello with remote mongo with:
```
foreman-installer \
  --scenario katello \
  --katello-pulp-db-username=pulp_user \
  --katello-pulp-db-password=pulp \
  --katello-pulp-db-seeds="192.168.121.43:27017" \
  --katello-pulp-db-name="pulp_database" \
  --foreman-admin-password=changeme
```

ekohl commented on manifests/pulp.pp:
```diff
@@ -54,6 +66,19 @@
+    db_ssl_certfile        => $db_ssl_certfile,
+    db_verify_ssl          => $db_verify_ssl,
+    db_ca_path             => $db_ca_path,
+    db_unsafe_autoretry    => $db_unsafe_autoretry,
```
This parameter isn't in the class parameters and thus causing a failure.

mbacovsky commented:
@ekohl, thanks, I've fixed the problem and updated the tests slightly.
I was thinking the same about the pulp module, but I'm not sure about the consequences.
Exposing whole pulp would lead to renamed existing installer parameters and answers keys which would be better to do prior this is merged though.

iNecas commented:
Given we target this for 1.18, I would like to shoot for less with this PR, even at the price of needed to change the parameters later. Since I don't expect too many uses using this option, handling the change by migration + updating docs later should be enough after we actually expose puppet-pulp, which should be tracked as a separate issue.
